Enable jemalloc on Mac

## Enhancement

Building TiFlash on Mac fails with jemalloc enabled:
`cmake .. -DENABLE_JEMALLOC=ON`

Error:
```
In file included from /Users/zanmato/dev/tiflash/contrib/jemalloc/src/bitmap.c:2:
In file included from /Users/zanmato/dev/tiflash/contrib/jemalloc/include/jemalloc/internal/jemalloc_internal_includes.h:51:
In file included from /Users/zanmato/dev/tiflash/contrib/jemalloc/include/jemalloc/internal/prof_structs.h:6:
/Users/zanmato/dev/tiflash/contrib/jemalloc/include/jemalloc/internal/mutex.h:41:4: error: unknown type name 'os_unfair_lock'
                        os_unfair_lock          lock;
                        ^
/Users/zanmato/dev/tiflash/contrib/jemalloc/include/jemalloc/internal/mutex.h:151:2: error: call to undeclared function 'os_unfair_lock_lock'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        MALLOC_MUTEX_LOCK(mutex);
        ^
/Users/zanmato/dev/tiflash/contrib/jemalloc/include/jemalloc/internal/mutex.h:83:37: note: expanded from macro 'MALLOC_MUTEX_LOCK'
#    define MALLOC_MUTEX_LOCK(m)    os_unfair_lock_lock(&(m)->lock)
                                    ^
/
```

--- a/contrib/jemalloc-cmake/include/jemalloc/internal/jemalloc_preamble.h
+++ b/contrib/jemalloc-cmake/include/jemalloc/internal/jemalloc_preamble.h
@@ -1,8 +1,8 @@
 #ifndef JEMALLOC_PREAMBLE_H
 #define JEMALLOC_PREAMBLE_H
 
-#include "jemalloc/internal/jemalloc_internal_decls.h"
 #include "jemalloc_internal_defs.h"
+#include "jemalloc/internal/jemalloc_internal_decls.h"
 
 #if defined(JEMALLOC_UTRACE) || defined(JEMALLOC_UTRACE_LABEL)
 #include <sys/ktrace.h>

--- a/libs/libcommon/cmake/find_jemalloc.cmake
+++ b/libs/libcommon/cmake/find_jemalloc.cmake
@@ -12,17 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Only enable under linux
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(ENABLE_JEMALLOC_DEFAULT 1)
-elseif(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
-    set(ENABLE_JEMALLOC_DEFAULT 0)
-elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    set(ENABLE_JEMALLOC_DEFAULT 0)
-endif()
-
-option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ${ENABLE_JEMALLOC_DEFAULT})
-# TODO: Make ENABLE_JEMALLOC_PROF default value to ${ENABLE_JEMALLOC_DEFAULT} after https://github.com/pingcap/tics/issues/3236 get fixed.
+option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ON)
+# TODO: Make ENABLE_JEMALLOC_PROF default value to ON after https://github.com/pingcap/tics/issues/3236 get fixed.
 option (ENABLE_JEMALLOC_PROF "Set to ON to enable jemalloc profiling" OFF)
 option (USE_INTERNAL_JEMALLOC_LIBRARY "Set to FALSE to use system jemalloc library instead of bundled" ${NOT_UNBUNDLED})
 

